### PR TITLE
integration[minor]: Llama Cpp streaming

### PR DESCRIPTION
**Pull request by nigel-daniels**
This PR adds streaming to the Llama CPP LLM and Chat modules.

### CODE REVIEW

File: langchain/src/chat_models/llama_cpp.ts [hunk 1]
```diff
@@ -128,6 +133,43 @@ export class ChatLlamaCpp extends SimpleChatModel<LlamaCppCallOptions> {
...
   }
 
+  async *_streamResponseChunks(
+    input: BaseMessage[],
+    // @ts-expect-error - TS6133: 'options' is declared but its value is never read.
+    options: this["ParsedCallOptions"],
+    runManager?: CallbackManagerForLLMRun
+  ): AsyncGenerator<ChatGenerationChunk> {
+    if (input.length !== 1) {
+      throw new Error("Only one human message should be provided.");
```
Comment (jacoblee93) [inline langchain/src/chat_models/llama_cpp.ts:143]: Is there a way we can fix this for multi-turn messages?

File: langchain/src/llms/llama_cpp.ts [hunk 2]
```diff
@@ -87,4 +89,29 @@ export class LlamaCpp extends LLM<LlamaCppCallOptions> {
...
   }
+
+  async *_streamResponseChunks(
+    prompt: string,
+    // @ts-expect-error - TS6133: 'options' is declared but its value is never read.
+    options: this["ParsedCallOptions"],
+    runManager?: CallbackManagerForLLMRun
+  ): AsyncGenerator<GenerationChunk> {
+    const promptOptions = {
+      temperature: this?.temperature,
```
Comment (jacoblee93) [inline langchain/src/llms/llama_cpp.ts:100]: maxTokens should be here too?

**Comment by jacoblee93** [comment]
Some small questions but I think good to go

**Comment by jacoblee93** [comment]
Thank you!

File: langchain/src/chat_models/llama_cpp.ts (hunk 1 above)
Comment (nigel-daniels) [reply langchain/src/chat_models/llama_cpp.ts:159]: I’ll explore this, ‘llama-node-cpp’ sets some of this up explicitly with the instantiation of a session. I should be able to compose a prompt annotated for ‘llama2’, although milage may vary if someone is using a none ‘llama2’ model.

File: langchain/src/llms/llama_cpp.ts (hunk 2 above)
Comment (nigel-daniels) [reply langchain/src/llms/llama_cpp.ts:98]: ‘maxTokens’ is set on the prompt method but to stream I’m using the evaluation method directly on the context. Unfortunately ‘maxTokens’ is not an option available on that method :(

**Pull request merged**